From e168a1973c3c3b956f419b4a93d8bb9db7fbf1ce Mon Sep 17 00:00:00 2001
From: Chaohong guo <chaohong.guo@intel.com>
Date: Wed, 23 May 2018 01:04:38 +0800
Subject: [RFC:test 4/7] Test script to download acrn hv/dm to local docker
 environment

the script downloaded ACRN hypervisor and device model code from github
into local docker disk, and prepare for building.

Signed-off-by: Chaohong guo <chaohong.guo@intel.com>
---
 test/3-clone-hv-dm.sh | 38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)
 create mode 100755 test/3-clone-hv-dm.sh

diff --git a/test/3-clone-hv-dm.sh b/test/3-clone-hv-dm.sh
new file mode 100755
index 0000000..3741e2c
--- /dev/null
+++ b/test/3-clone-hv-dm.sh
@@ -0,0 +1,38 @@
+#!/bin/bash
+#
+# This script is expected to run in ClearLinux host or docker developement
+# environment. Make sure system has the following commands before executing
+#      git
+#
+
+# export http_proxy=http://example.com:999
+# export https_proxy=http://example.com:999
+URL_ACRN=https://github.com/projectacrn/
+
+# This is the volume on host system, and will be mounted as volume into
+# docker -v ${HOST_DIR_VOL}:${ACRN_MNT_VOL}. We "git clone" all ACRN repos,
+# and then build disk image there in docker. Make sure that it has 30GB
+# space since we create disk image there.
+[ -z ${ACRN_MNT_VOL} ] && ACRN_MNT_VOL=/acrn-vol
+[ -z ${ACRN_ENV_VARS} ] && ACRN_ENV_VARS=acrn-env.txt
+cd ${ACRN_MNT_VOL} || { echo "Failed to cd "${ACRN_MNT_VOL}; exit -1; }
+
+[ -f ${ACRN_ENV_VARS} ] && \
+	                { for line in `cat ${ACRN_ENV_VARS}`; do export $line; done; }
+
+[ -z ${ACRN_HV_DIR} ] && ACRN_HV_DIR=acrn-hypervisor
+
+git clone ${URL_ACRN}/${ACRN_HV_DIR}
+
+if [ $? -ne 0 ]; then
+        echo "Failed to git-clone"
+        exit $?
+else
+        echo "Completed git-clone ACRN hypervisor and device model"
+fi;
+
+export ACRN_HV_DIR=${ACRN_HV_DIR}
+
+env | grep ACRN > ${ACRN_MNT_VOL}/${ACRN_ENV_VARS}
+
+exit 0;
-- 
2.14.1

